app/models/volunteer.rb

```diff
@@ -37,6 +37,22 @@ def case_assignments_with_cases
   def has_supervisor?
     supervisor_volunteer.present? && supervisor_volunteer.is_active?
   end
+
+  # false if volunteer has any case with no contact in the past 30 days
+  def made_contact_with_all_cases_in_days?(num_days = 30)
+    total_cases_count = casa_cases.count
+    return true if total_cases_count.zero?
+    current_contact_cases_count = cases_where_contact_made_in_days(num_days).count
+    return current_contact_cases_count == total_cases_count
+  end
+
+  private
+  def cases_where_contact_made_in_days(num_days = 30)
+    casa_cases
+      .joins(:case_contacts)
+      .where(case_contacts: { contact_made: true })
+      .where("case_contacts.occurred_at > ?", Date.today - num_days.days)
```

Comment:
I use `Date.current` per https://thoughtbot.com/blog/its-about-time-zones#a-summary-of-do39s-and-don39ts-with-time-zones